src/screens/JobProposal/SpecsView.test.tsx

```diff
@@ -136,16 +149,97 @@ describe('SpecsView', () => {
 
   describe('rejected proposal', () => {
     let specs: ReadonlyArray<JobProposal_SpecsFields>
+    let proposal: JobProposalPayloadFields
 
     beforeEach(() => {
+      proposal = buildJobProposal({ status: 'REJECTED' })
       specs = [buildJobProposalSpec({ status: 'REJECTED' })]
     })
 
     it('renders a rejected job proposal', async () => {
-      renderComponent(specs)
+      renderComponent(specs, proposal)
 
       expect(getByTestId('codeblock')).toHaveTextContent(specs[0].definition)
       expect(queryByText(/edit/i)).toBeNull()
     })
   })
+
+  describe('deleted proposal with approved spec', () => {
+    let specs: ReadonlyArray<JobProposal_SpecsFields>
+    let proposal: JobProposalPayloadFields
+
+    beforeEach(() => {
+      proposal = buildJobProposal({ status: 'DELETED' })
+      specs = [buildJobProposalSpec({ status: 'APPROVED' })]
+    })
+
+    it('renders a deleted job proposal', async () => {
+      renderComponent(specs, proposal)
+
+      expect(getByTestId('codeblock')).toHaveTextContent(specs[0].definition)
+      expect(queryByText(/edit/i)).toBeNull()
+    })
+
+    it('cancels the proposal', async () => {
+      renderComponent(specs, proposal)
+
+      userEvent.click(getByRole('button', { name: /cancel/i }))
```

Comment:
maybe I don't follow the style of test, but shouldn't the user be clicking confirm to cancel the proposal?

Reply:
the next line is the confirm button